vllm/entrypoints/tool_server.py

```diff
@@ -142,14 +150,16 @@ def get_tool_description(self, tool_name: str):
         return self.harmony_tool_descriptions.get(tool_name)
 
     @asynccontextmanager
-    async def new_session(self, tool_name: str):
+    async def new_session(self, tool_name: str, session_id: str):
         from mcp import ClientSession
         from mcp.client.sse import sse_client
         url = self.urls.get(tool_name)
+        headers = {"x-session-id": session_id}
         if not url:
             raise KeyError(f"Tool '{tool_name}' is not supported")
-        async with sse_client(url=url) as streams, ClientSession(
-                *streams) as session:
+        async with sse_client(url=url,
+                              headers=headers) as streams, ClientSession(
+                                  *streams) as session:
             await session.initialize()
             yield session
```

Comment:
For the session cleanup logic, maybe just add it here?

Reply:
Right now tool sessions are not managed in tool sever lifecycle but with context lifecycle so it has to be attached there unless we do a refactor

Reply:
But I think regardless of `exit_stack.push_async_exit(context.cleanup_session)` or implementing the exist logic here, the cleanup will be called when the code goes out the `async with AsyncExitStack() as exit_stack:` block?

And I thought different requests should have difference session so tool sessions should not be managed in tool server lifecycle. Is this understanding correct?

Reply:
This is in tool server, everything should be managed in context as it as the actual management of sessions and per request based